### PR TITLE
Added find files endpoint

### PR DESCRIPTION
**Pull request by VLiamin**

### CODE REVIEW

File: src/ProjectService.Broker/Requests/FileService.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    public FileService(
+      ILogger<FileService> logger,
+      IRequestClient<IGetFilesRequest> rcGetFiles)
+    {
+      _logger = logger;
+      _rcGetFiles = rcGetFiles;
+    }
+
+    public async Task<List<FileCharacteristicsData>> GetFilesAsync(List<Guid> filesIds, List<string> errors = null)
```
Comment (MariyaSin) [inline src/ProjectService.Broker/Requests/FileService.cs:28]: GetFilesCharacteristicsAsync

File: src/ProjectService.Models.Dto/Models/FileInfo.cs [hunk 2]
```diff
@@ -1,17 +1,12 @@
...
-    [Required]
-    public string Name { get; set; }
-    [Required]
-    public string Content { get; set; }
-    [Required]
+    public Guid Id { get; set; }
     public string Extension { get; set; }
-    public FileAccessType Access { get; set; }
+    public DateTime CreatedAtUtc { get; set; }
+    public int Size { get; set; }
```
Comment (MariyaSin) [inline src/ProjectService.Models.Dto/Models/FileInfo.cs:10]: add name

Comment (VLiamin) [reply src/ProjectService.Models.Dto/Models/FileInfo.cs:10]: removed this model.

File: src/ProjectService.Broker/Requests/FileService.cs (hunk 1 above)
Comment (VLiamin) [reply src/ProjectService.Broker/Requests/FileService.cs:28]: done

File: src/ProjectService.Data/FileRepository.cs [hunk 3]
```diff
@@ -30,11 +32,29 @@ public async Task<List<Guid>> CreateAsync(List<DbProjectFile> files)
...
+        return (null, 0);
+      }
+
+      IQueryable<DbProjectFile> dbFilesQuery = _provider.ProjectsFiles
+        .AsQueryable();
+
+      int totalCount = await dbFilesQuery.CountAsync();
+
+      List<DbProjectFile> dbFiles = await dbFilesQuery.Where(file => file.ProjectId == projectId && file.Access >= (int)access)
+        .Skip(filter.SkipCount).Take(filter.TakeCount).ToListAsync();
```
[the remark this answers is not in the document]
Comment (VLiamin) [reply src/ProjectService.Data/FileRepository.cs]: done

File: src/ProjectService.Business/Commands/File/FindFilesCommand.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    private readonly IResponseCreator _responseCreator;
+    private readonly IAccessValidator _accessValidator;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IProjectUserRepository _projectUserRepository;
+    private readonly IFileService _fileService;
+    private readonly IBaseFindFilterValidator _findFilterValidator;
+
+    public FindFilesCommand(
+      IFileRepository repository,
+      IFileAccessMapper accessMapper,
```
Comment (ndoriomedov) [inline src/ProjectService.Business/Commands/File/FindFilesCommand.cs:37]: do u use it?

File: src/ProjectService.Data/FileRepository.cs [hunk 5]
```diff
@@ -30,11 +32,27 @@ public async Task<List<Guid>> CreateAsync(List<DbProjectFile> files)
       await _provider.SaveAsync();
 
       return files.Select(x => x.FileId).ToList();
+    }
+
+    public async Task<(List<DbProjectFile>, int filesCount)> FindAsync(Guid projectId, BaseFindFilter filter, FileAccessType access = FileAccessType.Manager)
+    {
+      if (filter == null)
```
Comment (ndoriomedov) [inline src/ProjectService.Data/FileRepository.cs:39]: is null

**Review by ndoriomedov — APPROVED** [review]

File: src/ProjectService.Data/FileRepository.cs (hunk 5 above)
Comment (VLiamin) [reply src/ProjectService.Data/FileRepository.cs]: done

File: src/ProjectService.Business/Commands/File/FindFilesCommand.cs (hunk 4 above)
Comment (VLiamin) [reply src/ProjectService.Business/Commands/File/FindFilesCommand.cs]: done

File: src/ProjectService.Business/Commands/File/FindFilesCommand.cs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    public async Task<FindResultResponse<FileCharacteristicsData>> ExecuteAsync(FindProjectFilesFilter findFilter)
+    {
+      if (!_findFilterValidator.ValidateCustom(findFilter, out List<string> errors))
+      {
+        return _responseCreator.CreateFailureFindResponse<FileCharacteristicsData>(HttpStatusCode.BadRequest, errors);
+      }
+
+      FileAccessType accessType = FileAccessType.Public;
+
+      DbProjectUser user = (await _projectUserRepository.GetAsync(new List<Guid> { _httpContextAccessor.HttpContext.GetUserId() })).FirstOrDefault();
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/File/FindFilesCommand.cs:61]: add empty line

Comment (MariyaSin) [inline src/ProjectService.Business/Commands/File/FindFilesCommand.cs:61]: do you get all the user projects?
you should get user project by request project id and check the user role

File: src/ProjectService.Data/FileRepository.cs [hunk 7]
```diff
@@ -30,11 +32,27 @@ public async Task<List<Guid>> CreateAsync(List<DbProjectFile> files)
       await _provider.SaveAsync();
 
       return files.Select(x => x.FileId).ToList();
+    }
+
+    public async Task<(List<DbProjectFile>, int filesCount)> FindAsync(Guid projectId, BaseFindFilter filter, FileAccessType access = FileAccessType.Manager)
```
Comment (MariyaSin) [inline src/ProjectService.Data/FileRepository.cs:37]: why do you not use FindProjectFilesFilter?

File: src/ProjectService.Business/Commands/File/FindFilesCommand.cs (hunk 6 above)
Comment (VLiamin) [reply src/ProjectService.Business/Commands/File/FindFilesCommand.cs]: done

File: src/ProjectService.Data/FileRepository.cs (hunk 7 above)
Comment (VLiamin) [reply src/ProjectService.Data/FileRepository.cs]: done

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**